kfk.c

```diff
@@ -514,21 +514,19 @@ EXP K3(kfkSub){
   rd_kafka_resp_err_t err;
   rd_kafka_t *rk;rd_kafka_topic_partition_list_t *t_partition;
   J i;
-  I*p;
-  if(!checkType("is[I!]", x, y, z))
+  if(!checkType("i[sS][I!]", x, y, z))
     return KNL;
   if(!(rk= clientIndex(x)))
     return KNL;
   if(KFK_OK != (err = rd_kafka_subscription(rk, &t_partition)))
     return krr((S)rd_kafka_err2str(err));
-  if(z->t == XD){
```

Comment:
I think we're right to remove this but I think it will remove back compatibility with some earlier versions of the interface, if we are going to remove it then the final parameter probably should be deprecated fully as it is now not used at all